compiler/src/iree/compiler/Dialect/HAL/Target/TargetBackend.h

```diff
@@ -250,6 +250,8 @@ class TargetBackend {
     std::string dumpIntermediatesPath;
     // Optional path to write serialized binary results into.
     std::string dumpBinariesPath;
+    // Optional pass plugins
+    std::vector<std::string> passPlugins;
```

Comment:
The flags here are applied to all HAL targets - for target-specific flags they should be in the plugins. Since not all targets use LLVM and the user may want to apply different sets of plugins per target (CUDA vs CPU etc) it makes more sense to have them per-target. `ROCmOptions` has the options that are rocm-specific and you can add an `iree-hip-pass-plugins` option there. It'd be good to have it for other targets as well if the feature is useful (`iree-cuda-pass-plugins`, `iree-llvmcpu-pass-plugins`, etc)